server/application/application.go

```diff
@@ -245,9 +245,15 @@ func (s *Server) getApplicationEnforceRBACClient(ctx context.Context, action, pr
 		if !s.isNamespaceEnabled(namespaceOrDefault) {
 			return nil, security.NamespaceNotPermittedError(namespaceOrDefault)
 		}
-		return s.appclientset.ArgoprojV1alpha1().Applications(namespaceOrDefault).Get(ctx, name, metav1.GetOptions{
+		app, err := s.appclientset.ArgoprojV1alpha1().Applications(namespaceOrDefault).Get(ctx, name, metav1.GetOptions{
 			ResourceVersion: resourceVersion,
 		})
+		if err != nil {
+			return nil, err
+		}
+		// Objects returned by the lister must be treated as read-only.
+		// To allow us to modify the app later, make a copy
+		return app.DeepCopy(), nil
```

Comment:
Another example of - https://github.com/argoproj/argo-cd/pull/20805/files

We have to treat any objects returned as [read-only](https://github.com/kubernetes/code-generator/commit/87245b6442ebf48d87b67d31eb0418677836ba8a#diff-d645bf43d9617f9eb468a909d7aa9ae284f22ec45db4204ec7b33c631ce9c86chttps://github.com/kubernetes/code-generator/commit/87245b6442ebf48d87b67d31eb0418677836ba8a#diff-d645bf43d9617f9eb468a909d7aa9ae284f22ec45db4204ec7b33c631ce9c86c) but the application is modified in `inferResourcesStatusHealth`